proposals/0010-vk-buffer-ref.md

```diff
@@ -57,16 +57,16 @@ This new type will have the following operations
 
 * Copy assignment and copy construction - These copy the value of the pointer from one variable to another.
 * Dereference Method - The Get() method represents the struct const lvalue reference of the pointer to which it is applied. The selection . operator can be applied to the Get() to further select a member from the referenced struct.
-* Null Pointer Method - The IsNull() method returns true if the pointer is 0, false if not.
+* A new cast operator vk::pointer_cast<T,A>(p) is introduced. It creates a value of type BufferPointer<T,A>. The argument p can be of any BufferPointer type or uint64_t.
```

Comment:
I think we probably want to break this into two casts:
* `vk::static_pointer_cast<T, A>` allows casting any `vk::BufferPointer<SrcType, SrcAlign>` to `vk::BufferPointer<DstType, DstAlign>` only if `SrcType` is a type derived from `DstType`, and `DstAlign` >= `SrcAlign`.
* `vk::reinterpret_pointer_cast<T, A>` allows casting any `vk::BufferPointer<SrcType, SrcAlign>` to `vk::BufferPointer<DstType, DstAlign>` if `DstAlign` >= `SrcAlign`.

I don't think we should support casting `uint64_t` to `vk::BufferPointer` when we support construction instead, so the syntax `vk::BufferPointer<T,A>(MyUInt)` should suffice.

Reply:
I am generally fine with this. However, don't we want to require DstAlign <= SrcAlign? We know that a pointer aligned on 16 is also aligned on 8, but the reverse may not be true.

Reply:
OK. Will update with the alignment change.